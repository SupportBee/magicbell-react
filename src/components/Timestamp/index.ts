import Timestamp from './Timestamp';
export type { TimestampProps } from './Timestamp';
export default Timestamp;
