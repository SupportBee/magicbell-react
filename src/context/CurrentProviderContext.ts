import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
