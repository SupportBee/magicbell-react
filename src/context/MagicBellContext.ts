import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
