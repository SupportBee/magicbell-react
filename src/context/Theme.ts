export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
