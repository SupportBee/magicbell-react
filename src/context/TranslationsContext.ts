import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
