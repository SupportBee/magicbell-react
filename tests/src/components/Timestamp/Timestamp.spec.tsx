import '@testing-library/jest-dom/extend-expect';
import { act, render, RenderResult, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Sinon from 'sinon';
import Timestamp from '../../../../src/components/Timestamp';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';

describe('components', () => {
  describe('Timestamp', () => {
    let clock: Sinon.SinonFakeTimers;
    let view: RenderResult;

    beforeEach(() => {
      clock = Sinon.useFakeTimers(1600599000000);
      const date = new Date();
      clock.tick(60_000);

      view = render(
        <MagicBellThemeProvider value={defaultTheme}>
          <Timestamp date={date} />
        </MagicBellThemeProvider>,
      );
    });

    afterEach(() => {
      view.unmount();
      clock.restore();
    });

    describe('render', () => {
      it('renders the relative date', () => {
        expect(view.container).toMatchSnapshot();
      });

      it('updates the relative date text', () => {
        act(() => {
          clock.tick(120_000);
        });

        expect(view.container).toHaveTextContent('3m');
      });
    });

    describe('.onHover', () => {
      it('renders a tooltip', async () => {
        userEvent.hover(view.getByText('1m'));

        await waitFor(() => expect(view.container).toMatchSnapshot());
      });

      describe('the tooltipPlacement property is set to "left-end"', () => {
        beforeEach(() => {
          view.unmount();
          view = render(
            <MagicBellThemeProvider value={defaultTheme}>
              <Timestamp date={new Date()} tooltipPlacement="left-end" />
            </MagicBellThemeProvider>,
          );
        });

        it('renders a menu in the specified position', async () => {
          userEvent.hover(view.getByText('0s'));

          expect(await view.findByText('September 20, 2020 6:51 AM')).toBeInTheDocument();
        });
      });
    });
  });
});
