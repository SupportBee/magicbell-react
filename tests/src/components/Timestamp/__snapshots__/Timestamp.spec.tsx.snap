// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Timestamp .onHover renders a tooltip 1`] = `
.emotion-0 {
  background: #FFFFFF!important;
  border-radius: 8px!important;
  box-shadow: 2px 2px 3px rgba(230, 230, 230, 0.55);
  color: #3A424D!important;
  font-size: 0.8em!important;
  padding: 0.5em!important;
  text-transform: none;
}

<div>
  <div
    aria-expanded="false"
  >
    1m
  </div>
  <div
    data-tippy-root=""
    id="tippy-3"
    style="z-index: 9999; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px;"
  >
    <span
      class="emotion-0"
    >
      September 20, 2020 6:50 AM
    </span>
  </div>
</div>
`;

exports[`components Timestamp render renders the relative date 1`] = `
<div>
  <div
    aria-expanded="false"
  >
    1m
  </div>
</div>
`;
